From 8c0dcb4d9e39a3af0d0fcfb569e9b3dfa9d8b751 Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Mon, 18 Aug 2014 19:56:15 -0400
Subject: [scsi] qla2xxx: Replace constant value for IOCTL IOCB abort execution status with a macro for ISPFX00

Message-id: <1408391817-6389-28-git-send-email-cdupuis@redhat.com>
Patchwork-id: 87926
O-Subject: [PATCH RHEL 7.1 e-stor 27/69] qla2xxx: Replace constant value for IOCTL IOCB abort execution status with a macro for ISPFX00.
Bugzilla: 1089346
RH-Acked-by: Rob Evers <revers@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Armen Baloyan <armen.baloyan@qlogic.com>

scsi commit id b593931d6909d45bd186d53409808d3b38d20c6d

Signed-off-by: Armen Baloyan <armen.baloyan@qlogic.com>
Signed-off-by: Saurav Kashyap <saurav.kashyap@qlogic.com>
Signed-off-by: James Bottomley <JBottomley@Parallels.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/qla2xxx/qla_mr.c b/drivers/scsi/qla2xxx/qla_mr.c
index 89cfe56..f897d05 100644
--- a/drivers/scsi/qla2xxx/qla_mr.c
+++ b/drivers/scsi/qla2xxx/qla_mr.c
@@ -2017,7 +2017,8 @@ qlafx00_fx_disc(scsi_qla_host_t *vha, fc_port_t *fcport, uint16_t fx_type)
   memcpy(vha->hw->gid_list, pinfo, QLAFX00_TGT_NODE_LIST_SIZE);
  } else if (fx_type == FXDISC_ABORT_IOCTL)
   fdisc->u.fxiocb.result =
-      (fdisc->u.fxiocb.result == cpu_to_le32(0x68)) ?
+      (fdisc->u.fxiocb.result ==
+   cpu_to_le32(QLAFX00_IOCTL_ICOB_ABORT_SUCCESS)) ?
       cpu_to_le32(QLA_SUCCESS) : cpu_to_le32(QLA_FUNCTION_FAILED);
 
  rval = le32_to_cpu(fdisc->u.fxiocb.result);
diff --git a/drivers/scsi/qla2xxx/qla_mr.h b/drivers/scsi/qla2xxx/qla_mr.h
index 6cd7072..5f3ed8b 100644
--- a/drivers/scsi/qla2xxx/qla_mr.h
+++ b/drivers/scsi/qla2xxx/qla_mr.h
@@ -548,4 +548,7 @@ struct mr_data_fx00 {
 /* Max conncurrent IOs that can be queued */
 #define QLAFX00_MAX_CANQUEUE  1024
 
+/* IOCTL IOCB abort success */
+#define QLAFX00_IOCTL_ICOB_ABORT_SUCCESS 0x68
+
 #endif
-- 
1.7.1